product/manager.go

```diff
@@ -72,6 +73,8 @@ type manager struct {
 	productsMux      productsMux
 	cancelPolling    context.CancelFunc
 	prometheusLabels prometheus.Labels
+	etag             string
```

Comment:
Might it be better to put these definitions inside the pollingClosure() function (outside of thee return function).  There should be no need to use a mutex then.

Reply:
Good idea! Thanks.